src/dotnet/commands/dotnet-cache/CacheCommand.cs

```diff
@@ -0,0 +1,138 @@
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools.MSBuild;
+using Microsoft.DotNet.Tools.Restore;
+using Microsoft.DotNet.Cli.CommandLine;
+
+namespace Microsoft.DotNet.Tools.Cache
+{
+    public partial class CacheCommand
+    {
+        public string ProjectPath { get; set; }
+        public string Framework { get; set; }
+        public string Runtime { get; set; }
+        public string OutputPath { get; set; }
+        public string FrameworkVersion { get; set; }
+        public string IntermediateDir { get; set; }
+        public string Verbosity { get; set; }
+        private bool SkipOptimization { get; set; }
+        private bool PreserveIntermediateDir { get; set; }
+
+        public List<string> ExtraMSBuildArguments { get; set; }
+
+        private CacheCommand()
+        {
+        }
+
+        private void PrepareForExecute(CommandOption argument)
+        {
+            string targetProjectPath = string.Empty;
+
+            if (argument == null || !argument.Values.Any())
+            {
+                throw new InvalidOperationException(LocalizableStrings.SpecifyEntries);
+            }
+            try
+            {
+                if (string.IsNullOrEmpty(IntermediateDir))
+                {
+                    var tempDir = Path.GetTempPath();
+                    var folder = Path.GetRandomFileName();
+                    IntermediateDir = Path.Combine(tempDir, folder);
+                }
+                else
+                {
+                    IntermediateDir = Path.GetFullPath(IntermediateDir);
+                }
+
+                if (Directory.Exists(IntermediateDir))
+                {
+                    throw new GracefulException(string.Format(LocalizableStrings.IntermediateDirExists, IntermediateDir));
+                }
+
+                Directory.CreateDirectory(IntermediateDir);
+                var givenProject = argument.Values.FirstOrDefault(); //TODO ensure multiple entries can be processed
+                var projectFile = Path.GetFileName(givenProject);
+                targetProjectPath = Path.Combine(IntermediateDir, projectFile);
+
+                File.Copy(givenProject, targetProjectPath);
+
+                ProjectPath = targetProjectPath;
+            }
+            catch (Exception e)
+            {
+                throw new GracefulException(e.ToString());
+            }
+        }
+        private void CleanupAfterExecute()
+        {
+            if (!PreserveIntermediateDir)
+            {
+                try
+                {
+                    Directory.Delete(IntermediateDir, true);
+                }
+                catch (System.Exception e)
+                {
+                    throw new GracefulException(e.ToString());
+                }
+            }
+        }
+
+        public int Execute()
+        {
+            int result = 0;
+            List<string> msbuildArgs = new List<string>();
+            if (!string.IsNullOrEmpty(ProjectPath))
```

Comment:
Is this correct? Passing this to msbuild before the invocation to PrepareToExecute? You modify this property there.